### ISSUE TITLE
ForbiddenImport: all-under imports are matched without their trailing star

### ISSUE THREAD

**Issue by loganrosen** [detekt/detekt#9564]
## Expected Behavior

`ForbiddenImport` should treat an all-under import (`import java.util.*`) as the wildcard import it is.

With this config:

```yaml
ForbiddenImport:
  forbiddenImports:
    - value: 'java.util.*'
```

I'd expect `import java.util.*` to be reported. It pulls in every symbol the pattern was written to forbid.

And with this config:

```yaml
ForbiddenImport:
  forbiddenImports:
    - value: 'java.util.*'
  allowedImports:
    - 'java.util.UUID'
```

I'd expect `import java.util.UUID` to be exempt, but `import java.util.UUID.*` to still be reported. Allowing one class shouldn't silently allow a wildcard import of all its members.

## Observed Behavior

Both cases fail open — no finding is produced.

1. `forbiddenImports: ['java.util.*']` does not report `import java.util.*`.
2. `allowedImports: ['java.util.UUID']` exempts `import java.util.UUID.*` as well as `import java.util.UUID`.

The root cause is the same for both. `KtImportDirective.importedFqName` drops the trailing star, so `import java.util.*` reports its name as `java.util`, which is indistinguishable from `import java.util` and does not match the anchored regex `java\.util\..*` produced by the `java.util.*` pattern.

There's also a smaller cosmetic consequence: writing the pattern without a star (`forbiddenImports: ['java.util']`) *does* match the star import, but the finding message renders it as ``The import `java.util` has been forbidden``, omitting the star that's actually in the source.

Worth noting `forbiddenImports: ['java.util']` matches *only* the star import — because matching is anchored, it does not match `import java.util.Date`. So the spelling that reads like "forbid this package" does close to the opposite of what it looks like.

## Steps to Reproduce

Against `main` at 6fa04a63c8, using `ForbiddenImport(...).lint(code, compile = false)`:

| `forbiddenImports` | `allowedImports` | source | findings |
|---|---|---|---|
| `java.util.*` | - | `import java.util.*` | **0** |
| `java.util.*` | - | `import java.util.Date` | 1 |
| `java.util.*` | `java.util.UUID` | `import java.util.UUID.*` | **0** |
| `java.util.*` | `java.util.UUID` | `import java.util.UUID` | 0 |
| `java.util` | - | `import java.util.*` | 1, message says `java.util` |

Rows 2 and 4 are controls and behave as expected.

## Context

I ran into this while working on #6088 (documenting the rule's wildcard semantics) — I couldn't write accurate docs for all-under imports without describing this as a limitation.

The second case looks like it will bite #8091 directly. That issue plans to forbid `org.jetbrains.kotlin.name.StandardClassIds` while allowing `Name`, `FqName`, `CallableId` and `ClassId` — exactly the forbid-broad/allow-narrow shape where a wildcard import of an allowed class slips through unreported.

The general fix is to match against the import as it's written, star included, for both lists. That makes `forbiddenImports: ['java.util.*']` work as written and stops `allowedImports: ['java.util.UUID']` from covering `import java.util.UUID.*`. It's a breaking change for anyone relying on the starless spelling, though `allowedImports` only exists in 2.0 alphas so far, and I couldn't find any public config using the starless form.

I have a branch with this implemented and tested and am happy to open a PR, but this overlaps with #6075, which is on the same 2.0.0 milestone and uses `ForbiddenImports` as its worked example — so I'd rather check what shape you want here before sending it.

## Your Environment

* Version of detekt used: `main` @ 6fa04a63c8 (2.0.0-alpha line)
* Version of Gradle used (if applicable): n/a
* Gradle scan link (add `--scan` option when running the gradle task): n/a
* Operating System and version: macOS
* Link to your project (if it's a public repository): n/a

**Comment by cortinico** [comment]
> ForbiddenImport:
>   forbiddenImports:
>     - value: 'java.util.*'

The problem is that those values specified inside `forbiddenImports` are glob patterns as also explained in our docs:

> List of imports, specified as glob patterns, that are forbidden. It is recommended to also specify a reason.

This is consistent with other glob patterns that we have in the config. 

Changing the behavior to match what you're requesting is going to be a breaking change.

However we should fix this IMHO:

> There's also a smaller cosmetic consequence: writing the pattern without a star (forbiddenImports: ['java.util']) does match the star import, but the finding message renders it as The import `java.util` has been forbidden, omitting the star that's actually in the source.

Specifically we should not match against `import java.util.*`.

**Comment by BraisGabin** [comment]
#9673 fixes this issue but now I'm thinking if we should use the "simple" glob matching or the package-aware one... I guess that it's fine to report `import java.util.UUID.*` if you forbid `java.util.*`. I guess that we could talk about it if someone complains about it.
